doc: shields: Update shields doc with missing features

erwango opened the pull request:
Shields subsystem provides several undocumented features (as mentioned in #19086)
Update doc.

Change is split in 2 commits as one part is valid for 2.0 and second part is only available in later versions.

erwango committed:
doc: shields: Update shields doc with board overriding and variants

Shields subsystem provides the possibility to override configuration
for boards and to define variants.

Reflect this in documentation.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

mike-scott reviewed (approved):
Thank you @erwango for putting this up.  LGTM.

avisconti reviewed (approved):
LGTM

dbkinder commented on doc/guides/porting/shields.rst:
```diff
@@ -26,6 +27,11 @@ These files provides shield configuration as follows:
   compilation.
 
+* **<shield>.conf**: This file defines values for Kconfig symbols that are
+  required for default shield configuration. Please note that, to ease usage
```
For clarity, how about changing this last sentence to something like this:
```
To ease use with applications, the default shield
configurations here should be consistent with those
in the :ref:`default_board_configuration`.
```

dbkinder commented on doc/guides/porting/shields.rst:
```diff
@@ -59,6 +65,20 @@ introduced overriding node element:
+Board specific shield configuration
+-----------------------------------
+
+It may happen that some modifications are needed to fit a shield to a particular
```
How about instead:
```
If modifications are needed
to fit a shield to a particular board, you
can override a shield description for
a specific board by adding board-overriding
files to a shield, as follows:
```

dbkinder commented on doc/guides/porting/shields.rst:
```diff
@@ -77,3 +97,40 @@ Alternatively, it could be set by default in a project's CMakeLists.txt:
+   ├── <shield_v2>.overlay
+   └── <shield_v2>.conf
+
+In that case shield particluar revision should be activated using the revision
```
How about (for clarity and fixing a misspelling)
```
In this case, a shield-particular revision name can be used:
```

dbkinder commented on doc/guides/porting/shields.rst:
```diff
@@ -77,3 +97,40 @@ Alternatively, it could be set by default in a project's CMakeLists.txt:
+     :shield: shield_v2
+     :goals: build
+
+Again, it is also possible to provide a board specific configuration to a
```
Change to:
```
You can also provide a board-specific configuration
to a specific shield revision:
```

dbkinder reviewed (changes requested):
some recommended clarity edits

erwango commented:
@dbkinder, updated thanks for review

erwango commented:
@dbkinder, can you review again? Thanks!

dbkinder commented on doc/guides/porting/shields.rst:
```diff
@@ -26,6 +27,11 @@ These files provides shield configuration as follows:
   compilation.
 
+* **<shield>.conf**: This file defines values for Kconfig symbols that are
+  required for default shield configuration. Please note that, to ease use with
```
While it's polite, we don't need "please" in our documentation.  Delete "Please note that, " and start the sentence with "To ease use ..."

dbkinder reviewed (changes requested):
One little tweak, otherwise, LTGM.

erwango committed:
doc: shields: Document shield variant board overriding

It is now possible to override a shield variant configuration
for a specific board. Get it documented.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

dbkinder reviewed (approved):
LGTM, thanks.
